### PR TITLE
Fix http error

### LINKED ISSUE #354: didn't work
Flow Launcher version: 1.7.1
OS Version: Microsoft Windows NT 10.0.19041.0
IntPtr Length: 8
x64: True

Python Path: 
Date: 02/20/2021 20:20:14
Exception:
System.UriFormatException: Invalid URI: The hostname could not be parsed.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at System.Uri..ctor(String uriString)
   at Flow.Launcher.Infrastructure.Http.Http.UpdateProxy(ProxyProperty property) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Http\Http.cs:line 61
   at Flow.Launcher.Infrastructure.Http.Http.set_Proxy(HttpProxy value) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Http\Http.cs:line 43
   at Flow.Launcher.App.<OnStartupAsync>b__14_0() in C:\projects\flow-launcher\Flow.Launcher\App.xaml.cs:line 66
   at Flow.Launcher.Infrastructure.Stopwatch.NormalAsync(String message, Func`1 action) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Stopwatch.cs:line 59
   at Flow.Launcher.App.OnStartupAsync(Object sender, StartupEventArgs e) in C:\projects\flow-launcher\Flow.Launcher\App.xaml.cs:line 50
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

[2021-02-20.txt](https://github.com/Flow-Launcher/Flow.Launcher/files/6014920/2021-02-20.txt)

### PR DESCRIPTION
**Pull request by taooceros**
fix #354

### CODE REVIEW

**Comment by taooceros** [comment]
Todo: move http proxy initialization earlier before plugin initialization.

**Comment by jjw24** [comment]
is this ready to go? @taooceros

**Comment by taooceros** [comment]
> Todo: move http proxy initialization earlier before plugin initialization. 

Let me finish this first. I just get my computer.

**Comment by taooceros** [comment]
Done, please review @jjw24

File: Flow.Launcher/App.xaml.cs
Comment (jjw24) [inline Flow.Launcher/App.xaml.cs:82]: lol you love your spaces dont you 😛

File: Flow.Launcher.Test/HttpTest.cs
Comment (jjw24) [inline Flow.Launcher.Test/HttpTest.cs:14]: can you give this a bit more description please, Given-When-Then

File: Flow.Launcher/App.xaml.cs
Comment (taooceros) [reply Flow.Launcher/App.xaml.cs]: ah 🤣🤣🤣 I love to separate different parts of code doing different with spaces🤣🤣🤣

File: Flow.Launcher.Test/HttpTest.cs
Comment (taooceros) [reply Flow.Launcher.Test/HttpTest.cs:14]: ok, I will add comment.

Comment (jjw24) [reply Flow.Launcher.Test/HttpTest.cs:14]: hey if it's ok, i will update it for you, sorry i need to be afk soon and i want to get this release ready, so when you are free have a look at 1.7.2 and approve if all good, i will release it later

Comment (taooceros) [reply Flow.Launcher.Test/HttpTest.cs:14]: Yeah sure, if you would like to do it will be great. I just have dinner so haven't finish that part.

Comment (taooceros) [reply Flow.Launcher.Test/HttpTest.cs:14]: I will approve 1.7.2 once this has been merged. 🥰

**Review by jjw24 — APPROVED** [review]

**Pull request merged**
